.github/workflows/ci_coverage.yml

```diff
@@ -0,0 +1,30 @@
+name: CI and Coverage
```

Comment:
Any particular reason to put this in its own workflow, rather than adding it as an additional job to https://github.com/envoyproxy/gateway/blob/main/.github/workflows/build_and_test.yaml?

Reply:
> Any particular reason to put this in its own workflow, rather than adding it as an additional job to https://github.com/envoyproxy/gateway/blob/main/.github/workflows/build_and_test.yaml?

I notice the build_and_test workflow, cause of it's for code coverage，i think one workflow should be simplest.
for the coverage function, i think add additional job to build_and_test.yaml is ok.

Reply:
as @skriss suggested, prefer moving the codecov step [here](https://github.com/envoyproxy/gateway/blob/f08f90b5753fff70486aa626765d03abda5e8f5b/.github/workflows/build_and_test.yaml#L32)

Reply:
ok, Completed changes